AmplifyPlugins/Core/AWSPluginsCore/Auth/Provider/AuthIdentityProvider.swift

```diff
@@ -0,0 +1,14 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+
+public protocol AuthIdentityProvider {
+    func getIdentityId() -> Result<String, AmplifyAuthError>
+
+    func getUserSub() -> Result<String, AmplifyAuthError>
```

Comment:
Remind me: "user sub" is a Cognito term? Why isn't our protocol named to indicate this is Cognito-specific?